guides/source/wishlists.md

```diff
@@ -0,0 +1,1545 @@
+  <%= link_to "Add to wishlist", sign_up_path %>
+<% end %>
+```
+
+## Managing Wishlists
+
+Next, we need to be able to view and manage our wishlists
+
+### Wishlists Controller
+
+Start by adding a route for wishlists at the top level:
+
+```ruby#9
+Rails.application.routes.draw do
+  # ...
+  resources :products do
+    resource :wishlist, only: [ :create ], module: :products
+    resources :subscribers, only: [ :create ]
+  end
+  resource :unsubscribe, only: [ :show ]
+
+  resources :wishlists
+```
+
+Then we can add the controller at `app/controllers/wishlists_controller.rb` with
+the following:
+
+```ruby
+class WishlistsController < ApplicationController
+  allow_unauthenticated_access only: %i[ show ]
+  before_action :set_wishlist, only: %i[ edit update destroy ]
+
+  def index
+    @wishlists = Current.user.wishlists
+  end
+
+  def show
+    @wishlist = Wishlist.find(params[:id])
+  end
+
+  def new
+    @wishlist = Wishlist.new
+  end
+
+  def create
+    @wishlist = Current.user.wishlists.new(wishlist_params)
+    if @wishlist.save
+      redirect_to @wishlist, status: :see_other, notice: "Your wishlist was created successfully."
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @wishlist.update(wishlist_params)
+      redirect_to @wishlist, status: :see_other, notice: "Your wishlist has been updated successfully."
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @wishlist.destroy
+    redirect_to wishlists_path, status: :see_other
+  end
+
+  private
+
+  def set_wishlist
+    @wishlist = Current.user.wishlists.find(params[:id])
+  end
+
+  def wishlist_params
+    params.expect(wishlist: [ :name ])
+  end
+end
+```
+
+This is a very standard controller with a couple important changes:
+
+- Actions are scoped to `Current.user.wishlists` so only the owner can create,
+  update, and delete their own wishlists
+- `show` is publicly accessible so wishlists can be shared and viewed by anyone
+
+### Wishlist Views
+
+Create the index view at `app/views/wishlists/index.html.erb`:
+
+```erb
+<h1>Your Wishlists</h1>
+<%= link_to "Create a wishlist", new_wishlist_path %>
+<%= render @wishlists %>
+```
+
+This renders the `_wishlist` partial so let's create that at
+`app/views/wishlists/_wishlist.html.erb`:
+
+```erb
+<div>
+  <%= link_to wishlist.name, wishlist %>
+</div>
+```
+
+Next let's create the `new` view at `app/views/wishlists/new.html.erb`:
+
+```erb
+<h1>New Wishlist</h1>
+<%= render "form", locals: {wishlist: @wishlist} %>
+```
+
+And the `edit` view at `app/views/wishlists/edit.html.erb`:
+
+```erb
+<h1>Edit Wishlist</h1>
+<%= render "form", locals: {wishlist: @wishlist} %>
+```
+
+Along with the `_form` partial at `app/views/wishlists/_form.html.erb`:
+
+```erb
+<%= form_with model: @wishlist do |form| %>
+  <% if form.object.errors.any? %>
+    <div><%= form.object.errors.full_messages.first %></div>
```

Comment:
I'm curious about first here. Is that how we've been doing on previous tutorials?

I usually would do .to_sentence but I guess you wan to show one error at a time?

Reply:
I can't remember, but I agree we should use `to_sentence` here.

```suggestion
    <div><%= form.object.errors.full_messages.to_sentence %></div>
```